source/_integrations/firmata.markdown

```diff
@@ -0,0 +1,153 @@
+---
+title: Firmata
+description: "Connect Arduino-compatible boards within Home Assistant"
+ha_category:
+  - DIY
+  - Binary Sensor
+  - Switch
+ha_release: "0.110"
+ha_iot_class: Local Push
+ha_config_flow: false
+ha_codeowners:
+  - '@DaAwesomeP'
+ha_domain: firmata
+redirect_from:
+  - /components/binary_sensor.firmata/
+  - /components/switch.firmata/
+---
+
+[Firmata](https://github.com/firmata/protocol) can be used to add digital inputs and outputs to Home Assistant. Currently, this component supports high/low digital inputs and outputs. This allows for buttons, switches, motion detectors, relay control, etc. The component can currently connect to a Firmata board via serial or serial over USB.
+
+The Firmata protocol is a standard protocol for microcontrollers. Most of these boards are support digital and analog inputs and outputs. [Arduino](https://www.arduino.cc/) and Arduino-compatible microcontroller development boards are the most popular boards to use with Firmata.
+
+There is currently support for the following device types within Home Assistant:
+
+- [Binary Sensor](#binary_sensor)
+- [Switch](#switches)
+
+## Configuration
+
+You will need to upload a [Firmata firmware](https://github.com/firmata/) to your board. Please upload any `StandardFirmata` sketch to your board; please refer to the [Arduino documentation](https://www.arduino.cc/en/Main/Howto) for further information for uploading sketches. Firmata is available in the Arduino Library Manager in the Arduino IDE. You may also use a [Firmata Express](https://github.com/MrYsLab/FirmataExpress).
+
+To integrate an Arduino boards with Home Assistant, add the following section to your `configuration.yaml` file:
+
+
+
+<div class='note warning'>
+Firmata does not store last state accross power cycles. This means that with every powerup the pins are set to off/low and after every initialization they will be set to the default that you have configured in YAML.
+</div>
+
+<div class='note warning'>
+Firmata may cycle pins HIGH/LOW during initialization. If your pins should not be accidentally cycled, you may need to modify your Firmtata sketch or if you are using a relay board use an output pin as VCC. See this issue for details: <a href="https://github.com/firmata/arduino/issues/166" target="_blank">Make initial pin value configurable</a>
```

Comment:
It's also possible to use the markdown style for the link to Github: [See our docs](https://developers.home-assistant.io/docs/documenting/create-page#html)

Reply:
I tried but it doesn't seem to work inside the `<div>`.

Reply:
Oh I see! I need a newline!